NameLayerMain/src/vg/civcraft/mc/namelayer/database/GroupManagerDao.java

```diff
@@ -820,51 +871,101 @@ public synchronized void removeSubGroup(String group, String subGroup){
 		}
 	}
 
-	public synchronized void addPermission(String groupName, String role, String values){
+	public synchronized void addPermission(String groupName, String role, List <PermissionType> perms){
 		NameLayerPlugin.reconnectAndReintializeStatements();
-		try {
-			addPerm.setString(1, role);
-			addPerm.setString(2, values);
-			addPerm.setString(3, groupName);
-			addPerm.execute();
-		} catch (SQLException e) {
-			plugin.getLogger().log(Level.WARNING, "Problem adding " + role + " with " + values
-					+ " to group " + groupName, e);
+		for(PermissionType perm : perms) {
+			try {
+				addPermission.setString(1, role);
+				addPermission.setInt(2, perm.getId());
+				addPermission.setString(3, groupName);
+				addPermission.execute();
+			} catch (SQLException e) {
+				plugin.getLogger().log(Level.WARNING, "Problem adding " + role + " with " + perms
+						+ " to group " + groupName, e);
+			
+			}
 		}
 	}
 	
 	public synchronized Map<PlayerType, List<PermissionType>> getPermissions(String group){
 		NameLayerPlugin.reconnectAndReintializeStatements();
 		Map<PlayerType, List<PermissionType>> perms = new HashMap<PlayerType, List<PermissionType>>();
 		try {
-			getPerms.setString(1, group);
-			ResultSet set = getPerms.executeQuery();
+			getPermission.setString(1, group);
+			ResultSet set = getPermission.executeQuery();
 			while(set.next()){
 				PlayerType type = PlayerType.getPlayerType(set.getString(1));
-				String length = set.getString(2);
-				String[] multiPerms = length.split(" ");
-				List<PermissionType> listPerm = new ArrayList<PermissionType>();
-				for (String x: multiPerms)
-					listPerm.add(PermissionType.getPermissionType(x));
-				perms.put(type, listPerm);
+				List<PermissionType> listPerm = perms.get(type);
+				if (listPerm == null) {
+					listPerm = new ArrayList<PermissionType>();
+					perms.put(type, listPerm);
+				}
+				int id = set.getInt(2);
+				PermissionType perm = PermissionType.getPermission(id);
+				if (perm != null) {
+					listPerm.add(perm);
+				}
 			}
 		} catch (SQLException e) {
 			plugin.getLogger().log(Level.WARNING, "Problem getting permissions for group " + group, e);
 		}
 		return perms;
 	}
 	
-	public synchronized void updatePermissions(String group, PlayerType pType, String perms){
+	public synchronized void updatePermissions(String group, PlayerType pType, List<PermissionType> perms){
```

Comment:
So for update you're just removing all and replacing in entire? 

Hm. This is definitely easier programmatically but not great in terms of DB resources, especially since addPermissions isn't batched and performs each playertype query independently